lib/Fmt_ast.ml

```diff
@@ -3270,20 +3299,29 @@ and fmt_class_field_kind c ctx = function
         match (e, args') with
         | {pexp_desc= Pexp_constraint (e, t); _}, [] ->
             Some (List.rev names, t, e)
-        | ( {pexp_desc= Pexp_newtype (({txt; _} as newtyp), body); _}
-          , {txt= txt'; _} :: args )
+        | ( { pexp_desc=
+                Pexp_newtype ((({txt; _} as new_name), new_layout), body)
+            ; _ }
+          , ({txt= txt'; _}, _) :: args )
           when String.equal txt txt' ->
-            cleanup (newtyp :: names) body args
+            cleanup ((new_name, new_layout) :: names) body args
         | _ -> None
       in
       match cleanup [] e poly_args with
       | Some (args, t, e) ->
-          let before =
-            match args with x :: _ -> x.loc | [] -> e.pexp_loc
+          let before_name, before_layout_opt =
+            match args with
+            | (x, y) :: _ -> (x.loc, Option.map ~f:(fun l -> l.loc) y)
+            | [] -> (e.pexp_loc, Some e.pexp_loc)
           in
-          Cmts.relocate c.cmts ~src:pexp_loc ~before ~after:e.pexp_loc ;
+          Cmts.relocate c.cmts ~src:pexp_loc ~before:before_name
+            ~after:e.pexp_loc ;
+          Option.iter
```

Comment:
This new `Cmts.relocate` supersedes the previous one, right? Then I think it would be cleaner to just have a `match before_layout_opt with ...` that either does one `relocate` call or the other. As it is, I've stared at this for 5 minutes and I'm still not sure exactly what's going on here.

Reply:
I think I over-complicated the logic here and there's no need for the call to `Cmts.relocate` inside the `Option.iter`. Relocating the comments to before the tyvar name should be enough.